plugins/modules/system/java_keystore.py

```diff
@@ -201,6 +207,11 @@ def create_jks(module, name, openssl_bin, keytool_bin, keystore_path, password):
 
             export_p12_cmd = "%s pkcs12 -export -name '%s' -in '%s' -inkey '%s' -out '%s' -passout 'pass:%s'" % (
                 openssl_bin, name, certificate_path, private_key_path, keystore_p12_path, password)
+
+            # when keypass is provided, add -passin
+            if keypass:
+                export_p12_cmd += " -passin 'pass:%s'" % keypass
```

Comment:
From a security point of view, this is really bad. The password can now be accessed by other users on the system by looking at the output of `ps aux`. Passwords should **never** be passed as command line arguments, but always provided via `stdin`.

Reply:
interesting whether the `run_command` provides this opportunity

Reply:
It does. The option is called `data`: https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/basic.py#L2393

Reply:
Ok and I believe the original password stored using -passout must also be provided using the "data" option?

Reply:
Hmm, did the module do that? That's not good, and yes, it should also be passed via stdin.

Reply:
Looking at it a bit more, I am not sure it is going to work. The data argument is a raw standard input for the process being executed, and with this tools "openssl" and "keytool" we would need to interactively wait for a prompt and provide the passwords.

Maybe I can write a temporary file for the respective passwords and use them as an argument. With openssl we can specify using, like: `-passin "file:/tmp/pass.tmp"` and with keytool, maybe we can try using `<(/tmp/file)` (although I am not sure if that helps).

Reply:
Let me know what you guys think. For now I have pushed another PR with arguments being passed using a list instead of a raw string.

Reply:
... and one more thing, the password it not being displayed in the output (using -vvv) because of the `no_log_strings`, so I am just seeing: `private_key_passphrase: VALUE_SPECIFIED_IN_NO_LOG_PARAMETER`.

Reply:
Passing multiple passwords via stdin should be fine too, since you know in which order they will be asked by the tool. Temp files are also not a good idea. What could be done are using named pipes. Though stdin is probably a lot simpler.

Reply:
@felixfontein I tried locally using setting a string with multiple lines (separated by \n) and passing it as the `data` argument to `run_command`, but it did not work. I will play with it tonight or tomorrow and see if I can get it to work. Otherwise, I will provide my findings here.
Thanks for the review.

Reply:
I was able to get passwords provided to openssl via stdin.